frontend/src/components/series/SeriesCard.tsx

```diff
@@ -1,18 +1,30 @@
-import {Series} from "../../models/Series";
+import {DepositSeries, PaymentSeries} from "../../models/Series";
 import MonetaryValue from "../MonetaryValue";
 import FormatDate from "../FormatDate";
 import Button from "../Button";
 
 interface ISeriesItemProps {
-    series: Series
-    deleteSeries: (seriesId?: string) => void
+    series: PaymentSeries | DepositSeries
+    deleteSeries: (seriesId: string | undefined, type: string) => void
 }
 
 export default function SeriesCard({series, deleteSeries}: ISeriesItemProps) {
 
+    const amount: number = ("payment" in series && series.payment.amount) ||
+        ("deposit" in series && series.deposit.amount)
+        || 0
+
+    const description: string = ("payment" in series && series.payment.description) ||
+        ("deposit" in series && series.deposit.description)
+        || ""
+
+    const type: string = ("payment" in series && "payment") ||
+        ("deposit" in series && "deposit")
+        || ""
+
```

Comment:
I would refactore this. You are checking three times if "payment"/"deposit" is in series, maybe make a check at the beginning which series you have and then get the values.